wandb/sdk/artifacts/artifact.py

```diff
@@ -501,6 +473,54 @@ def metadata(self, metadata: dict) -> None:
         """
         self._metadata = self._normalize_metadata(metadata)
 
+    @property
+    def ttl(self) -> Union[datetime.timedelta, ArtifactTTL, None]:
```

Comment:
@szymon-piechowicz-wandb so I have implemented some of what we talked about. 
1. we want to make sure the user isn't confused about what duration or enum to set and get back 
2. make sure we can correctly parse it in the backend

I implemented what I think works but took some shortcuts from what we talked about by using less flags/enums.
```
artifact.ttl = None sets _ttl_duration_seconds = INHERIT (-1), getter = None
artifact.ttl = DISABLED sets _ttl_duration_seconds = DISABLED (-2), getter = DISABLED
artifact.ttl = timedelta(seconds=10) sets _ttl_duration_seconds = 10, getter = 10
Nothing, _ttl_duration_seconds = None, getter = None
```

Updated flow in https://lucid.app/lucidchart/4dedff6a-7747-4849-819d-c152fd608f99/edit?invitationId=inv_c6538a29-c706-4179-9cb8-1192be3332ad&page=0_0#

Reply:
Also didn't add documentation for `ArtifactTTL.DISABLED` since it doesn't really do anything until artifact collection ttl comes out.